File: C/Easy(arrays)/second_largest_element.c
#include<stdio.h>

int main()
{
    int a[100], n;
    scanf("%d", &n);

    if (n < 2) {
        printf("Need at least two numbers\n");
        return 0;
    }

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &a[i]);
    }

    int max = a[0];
    int max_2 = a[0];

    for (int i = 1; i < n; i++)
    {
        if (a[i] > max)
        {
            max_2 = max;
            max = a[i];
        }
        else if (a[i] > max_2 && a[i] != max)
        {
            max_2 = a[i];
        }
    }

    if (max_2 == max)
    {
        printf("max = %d\n", max);
        printf("There is no second distinct maximum\n");
    }
    else
    {
        printf("max = %d\n", max);
        printf("max_2 = %d\n", max_2);
    }

    return 0;
}

File: Python/Easy(Lists)/second_largest_element.py
a = []
n = int(input())

for _ in range(n):
    a.append(int(input()))

if n < 2:
    print("Need at least two numbers")
else:
    max_value = a[0]
    max_value_2 = float("-inf")

    for value in a[1:]:
        if value > max_value:
            max_value_2 = max_value
            max_value = value
        elif value > max_value_2 and value != max_value:
            max_value_2 = value

    if max_value_2 == float("-inf"):
        print(f"The max value is {max_value}")
        print("There is no second distinct maximum")
    else:
        print(f"The max value is {max_value}")
        print(f"The 2nd max value is {max_value_2}")